data_pipelines_cli/cli_commands/generate.py

```diff
@@ -0,0 +1,300 @@
+@click.group(name="generate", help="Generate additional dbt files")
+def generate_group() -> None:
+    pass
+
+
+@generate_group.command(
+    name="model-yaml", help="Generate schema YAML using codegen or dbt-profiler"
+)
+@click.option("--env", default="local", type=str, help="Name of the environment", show_default=True)
+@click.option(
+    "--with-meta", type=bool, is_flag=True, help="Whether to generate dbt-profiler metadata"
+)
+@click.option(
+    "--overwrite", type=bool, is_flag=True, help="Whether to overwrite existing YAML files"
+)
+@click.argument(
+    "model-path",
+    type=click.Path(exists=True, path_type=pathlib.Path, file_okay=False, dir_okay=True),
+    nargs=-1,
+)
+def _generate_model_yaml_command(
+    env: str, with_meta: bool, overwrite: bool, model_path: Sequence[pathlib.Path]
+) -> None:
+    if len(model_path) == 0:
+        raise DataPipelinesError("Command expects at least one 'model-path' argument")
+    generate_model_yamls(env, with_meta, overwrite, model_path)
+
+
+@generate_group.command(name="source-yaml", help="Generate source YAML using codegen")
+@click.option("--env", default="local", type=str, help="Name of the environment", show_default=True)
+@click.option(
+    "--source-path",
+    default=pathlib.Path.cwd().joinpath("models", "source"),
+    show_default=True,
+    type=pathlib.Path,
+    help="Path to the 'source' directory",
+    required=True,
+)
+@click.option(
+    "--overwrite", type=bool, is_flag=True, help="Whether to overwrite an existing YAML file"
+)
+@click.argument("schema-name", type=str, nargs=-1)
+def _generate_source_command(
+    env: str, source_path: pathlib.Path, overwrite: bool, schema_name: Sequence[str]
+) -> None:
+    if len(schema_name) == 0:
+        raise DataPipelinesError("Command expects at least one 'schema-name' argument")
+    generate_source_yamls(env, source_path, overwrite, schema_name)
+
+
+@generate_group.command(
+    name="source-sql", help="Generate SQLs that represents tables in given dataset"
+)
+@click.option("--env", default="local", type=str, help="Name of the environment", show_default=True)
+@click.option(
+    "--source-yaml-path",
+    type=click.Path(exists=True, path_type=pathlib.Path, file_okay=True),
+    default=pathlib.Path.cwd().joinpath("models", "source", "source.yml"),
+    show_default=True,
+    help="Path to the 'source.yml' schema file",
+    required=True,
+)
+@click.option(
+    "--staging-path",
+    default=pathlib.Path.cwd().joinpath("models", "staging"),
+    show_default=True,
+    type=pathlib.Path,
+    help="Path to the 'staging' directory",
+    required=True,
+)
+@click.option(
+    "--overwrite", type=bool, is_flag=True, help="Whether to overwrite existing SQL files"
+)
+def _generate_source_sqls(
+    env: str, source_yaml_path: pathlib.Path, staging_path: pathlib.Path, overwrite: bool
+) -> None:
+    generate_source_sqls(env, source_yaml_path, staging_path, overwrite)
```

Comment:
Could you please break this file into smaller? 300 lines of code it 3 times tooo long ;)

Reply:
Split